File: src/domain/label/entities/LabelItem.js
/* exported LabelItem */

class LabelItem {
  /**
   * @param {string} sku
   * @param {number} quantity
   */
  constructor(sku, quantity) {
    this.sku = String(sku || '').trim();
    this.quantity = Number(quantity || 0) || 0;

    if (!this.sku) {
      throw new Error('SKUが空です');
    }
    if (!Number.isFinite(this.quantity) || this.quantity <= 0) {
      throw new Error(`数量が不正です: ${quantity}`);
    }
  }

  /**
   * Downloader.downloadLabels() の mskuQuantities 形式へ変換
   * @returns {{msku: string, quantity: number}}
   */
  toMskuQuantity() {
    return { msku: this.sku, quantity: this.quantity };
  }
}

File: python/src/domain/label/entities/label_item.py
from __future__ import annotations


class LabelItem:
    def __init__(self, sku: str, quantity: int) -> None:
        self.sku = str(sku or "").strip()
        self.quantity = int(quantity) if quantity else 0
        if not self.sku:
            raise ValueError("SKUが空です")
        if self.quantity <= 0:
            raise ValueError(f"数量が不正です: {quantity}")

    def to_msku_quantity(self) -> dict[str, str | int]:
        return {"msku": self.sku, "quantity": self.quantity}
